pyro/ops/stats.py

```diff
@@ -1,30 +1,36 @@
 from __future__ import absolute_import, division, print_function
 
+import math
+import numbers
+
 import torch
 
+from pyro.distributions.util import logsumexp
+
 
 def _compute_chain_variance_stats(input):
     # compute within-chain variance and variance estimator
     # input has shape N x C x sample_shape
     N = input.size(0)
-    chain_mean = input.mean(dim=0)
-    var_between = chain_mean.var(dim=0)
-
     chain_var = input.var(dim=0)
     var_within = chain_var.mean(dim=0)
-    var_estimator = (N - 1) / N * var_within + var_between
+    var_estimator = (N - 1) / N * var_within
+    if input.size(1) > 1:
```

Comment:
Will this ever be false, since it seems that we are asserting below that `input.size(chain_dim) >= 2`?

Reply:
@neerajprad Both `split_gelman_rubin` and `effective_sample_size` don't require `input.size(chain_dim) >= 2`. So we can get `diagnostics` in HMC even with `num_chains=1`.